### ISSUE TITLE
[Bug]: You can go to any checkout step without previous validation

### ISSUE THREAD

**Issue by bartoszherba** [vuestorefront/magento2#409]
### Expected Behavior
Whenever I visit checkout I can go through or go to a particular step only if previous step is completed and validated.

### Actual Behavior
Once I land on checkout I can go to any step without validation of previous steps, eg. I can instantly go to the last step and try to place an order.

### Possible Solution
Checkout navigation must follow rules:

* I can go to the next step only if the previous step is already completed and validated
* I can revisit any previous step at any time
* If I revisit one of the previous steps and change any data I will have to revalidate all next steps again (as previous data might have an impact on further calculation)

### Steps to reproduce
Simply go to the checkout and instantly switch to the last step. You can even try to place an order without previously validated steps.

### What version of Magento 2 integration are you using?
2.4.x

### What version of Node.js are you using?
16.13.1

### What browser (and version) are you using?
Version 96.0.4664.110 (Official Build) (arm64)

### What operating system (and version) are you using?
macOS Monterey 12.1

### Code of Conduct
* [x] I agree to follow this project's Code of Conduct

┆Attachments: <a href="https://vsf.atlassian.net/rest/api/2/attachment/content/11089">image-20220125-133016.png</a>

**Comment by Frodigo** [comment]
@igorwojciechowski thanks for catching this. We have separate ticket for that: https://github.com/vuestorefront/magento2/issues/492
